src/marten/template/filter/join.cr

```diff
@@ -0,0 +1,20 @@
+module Marten
+  module Template
+    module Filter
+      # The "join" filter.
+      #
+      # The "join" filter converts an array into a string with elements separated by `arg`.
+      class Join < Base
+        def apply(value : Value, arg : Value? = nil) : Value
+          # Only work on array
+          return value if !value.raw.is_a?(Array)
```

Comment:
I think we'll want to ensure that the `join` filter can be used for all [`Enumerable`](https://crystal-lang.org/api/1.7.2/Enumerable.html) objects (because all of these could respond to `#to_a`). Moreover, for filters that work only for specific types of values, we usually raise an error to indicate that the filter was misused!

For example:

```crystal
def apply(value : Value, arg : Value? = nil) : Value
  if !value.raw.is_a?(Enumerable)
    raise Errors::UnsupportedType.new("#{value.raw.class} objects can't be joined")
  end

  if arg
    Value.from(value.to_a.map(&.to_s).join(arg.to_s))
  else
    Value.from(value.to_a.map(&.to_s).join)
  end
end
```

Reply:
I think I've done this!